src/metaxy/metadata_store/base.py

```
@@ -117,6 +128,19 @@ def __init__(
         self._context_depth = 0
         self._prefer_native = prefer_native
 
+        # Resolve auto_create_tables from global config if not explicitly provided
+        if auto_create_tables is None:
+            try:
+                from metaxy.config import MetaxyConfig
+
+                config = MetaxyConfig()
+                self.auto_create_tables = config.auto_create_tables
+            except Exception:
```

no silent error handling!